yt_dlp/extractor/fifa.py

```diff
@@ -1,67 +1,244 @@
+            'title': "Le Sommer: Lyon-Barcelona a beautiful final for women's football",
         },
         'params': {'skip_download': 'm3u8'},
+        'skip': 'HTTP Error 403: Forbidden',
+    }, {
+        'url': 'https://www.fifa.com/fifaplus/en/watch/4V8H8qv7QM1LNVk5gUwYFa',
+        'info_dict': {
+            'id': '709abaec-5eef-4ad8-a02d-19a8932f42a2',
+            'title': "Christine Sinclair at 19 | FIFA U-19 Women's World Championship Canada 2002™",
+            'display_id': 'christine-sinclair-at-19-fifa-u-19-womens-world-championship-canada-2002',
+            'thumbnail': r're:https://cdn\.plus\.fifa\.com//images/public/cms/[/\w-]+\.jpg\?width=1408',
+        },
+        'params': {
+            'skip_download': 'm3u8',
+            'ignore_no_formats_error': True,
+        },
+        'expected_warnings': [
+            'Requested format is not available',
+            'This video is DRM protected',
+        ],
+    }, {
+        'url': 'https://www.fifa.com/fifaplus/en/watch/d85632f9-7009-4ea0-aaf1-8d6847e4a148',
+        'info_dict': {
+            'id': 'bbe5d2a3-3dfd-4283-a1af-3a66022e8254',
+            'title': 'Croatia v Australia | Group F | 2006 FIFA World Cup Germany™ | Full Match Replay',
+            'display_id': 'croatia-v-australia-or-group-f-or-2006-fifa-world-cup',
+            'thumbnail': r're:https://cdn\.plus\.fifa\.com//images/public/cms/[/\w-]+\.jpg\?width=1408',
+        },
+        'params': {
+            'skip_download': 'm3u8',
+            'ignore_no_formats_error': True,
+        },
+        'expected_warnings': [
+            'Requested format is not available',
+            'This video is DRM protected',
+        ],
+    }, {
+        'url': 'https://www.fifa.com/fifaplus/pt/watch/Ny88zzqsVnxCBUJ6fZzPy',
+        'info_dict': {
+            'id': '3d2612ff-c06f-4a7e-a2d7-ec73504515b5',
+            'title': 'The Happiest Man in the World',
+            'display_id': 'o-homem-mais-feliz-do-mundo',
+            'thumbnail': r're:https://cdn\.plus\.fifa\.com//images/public/cms/[/\w-]+\.jpg\?width=1408',
+        },
+        'params': {
+            'skip_download': 'm3u8',
+            'ignore_no_formats_error': True,
+        },
+        'expected_warnings': [
+            'Requested format is not available',
+            'This video is DRM protected',
+        ],
     }]
 
     def _real_extract(self, url):
         video_id, locale = self._match_valid_url(url).group('id', 'locale')
-        webpage = self._download_webpage(url, video_id)
 
-        preconnect_link = self._search_regex(
-            r'<link\b[^>]+\brel\s*=\s*"preconnect"[^>]+href\s*=\s*"([^"]+)"', webpage, 'Preconnect Link')
+        if redirect_url := traverse_obj(self._call_api(
+                f'pages/{locale}/watch/{video_id}', video_id, 'Downloading redirection info'), 'redirectUrl'):
+            return self.url_result(redirect_url)
```

Comment:
does the video ID change in the `redirectUrl`? can extraction not continue beyond this point if a `redirectUrl` exists? just curious how this works

Reply:
Example:

URL: https://www.fifa.com/fifaplus/en/watch/7on10qPcnyLajDDU3ntg6y, the `redirectUrl` is: https://www.plus.fifa.com/en/content/italy-v-france-final-2006-fifa-world-cup-germany-full-match-replay/fee2f7e8-92fa-42c5-805c-a2c949015eae

We can also get the URL by following HTTP redirection:

- Navigate to https://www.fifa.com/fifaplus/en/watch/7on10qPcnyLajDDU3ntg6y
- 301: https://www.fifa.com/en/watch/7on10qPcnyLajDDU3ntg6y
- 301: https://www.plus.fifa.com/en/content/italy-v-france-final-2006-fifa-world-cup-germany-full-match-replay/fee2f7e8-92fa-42c5-805c-a2c949015eae
- 200 done.

.